app/javascript/controllers/dropdate_picker_controller.js

```diff
@@ -0,0 +1,81 @@
+import { Controller } from "@hotwired/stimulus"
+import flatpickr from "flatpickr"
+
+export default class extends Controller {
+  static targets = ["input", "date", "time", "zone", "showing", "editing"]
+
+  connect() {
+    // localize inputs
+    if (this.inputTarget.value) {
+      const date = new Date(this.inputTarget.value)
+      this.setValue(this.dateTarget, date.toLocaleDateString(), true)
+      this.setValue(this.timeTarget, date.toLocaleTimeString(), true)
+    }
+
+    // listen for changes
+    this.dateTarget.dataset.action = `${this.dateTarget.dataset.action} dropdate-picker#change dropdate-picker#defaultTime`
+    this.timeTarget.dataset.action = `${this.timeTarget.dataset.action} dropdate-picker#change`
+    this.zoneTarget.dataset.action = `${this.zoneTarget.dataset.action} dropdate-picker#change`
+
+    // init flatpickr
+    flatpickr(this.dateTarget, {
+      allowInput: true,
+      dateFormat: "n/j/Y",
+    })
+    flatpickr(this.timeTarget, {
+      allowInput: true,
+      dateFormat: "h:i:S K",
+      enableTime: true,
+      enableSeconds: true,
+      noCalendar: true,
+    })
+  }
+
+  editing(event) {
+    event.preventDefault()
```

Comment:
just checking, but the default action here would be some sort of `submit`? because this `dropdate_picker#editing` action is used to toggle which fields the user sees (the showing field vs. the editing fields)

Reply:
I think the little edit-pencil is a `<a>` link, not a button.  So clicking it will by default try to navigate.  Which we don't want.